frontend/src/pages/matching/Matching.tsx

```diff
@@ -68,15 +69,29 @@ const Matching: React.FC = () => {
     setMatchingState(MatchingStateEnum.PENDING);
   };
 
+  const handleCancel: React.FormEventHandler<HTMLFormElement> = (e) => {
+    socket?.emit('disconnect');
```

Comment:
```suggestion
    e.preventDefault();
    socket?.disconnect();
```

prevent reloading page + make socket.io happy ([disconnect is reserved event name](https://socket.io/docs/v4/emit-cheatsheet/))